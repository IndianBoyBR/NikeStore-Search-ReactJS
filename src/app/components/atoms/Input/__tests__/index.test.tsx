import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import { Store } from 'redux';
import { configureAppStore } from 'store/configureStore';
import { Input } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const ComponentWithParentAndProviders = (
  props: Parameters<typeof Input>[number],
  store: Store,
) =>
  render(
    <Provider store={store}>
      <Input {...props} />
    </Provider>,
  );

describe('<Input />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });

  it('should match snapshot', () => {
    const { getByTestId, container } = ComponentWithParentAndProviders(
      {
        type: 'search',
        placeholder: 'placeholder',
      },
      store,
    );

    const searchElement = getByTestId('input');

    expect(searchElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
