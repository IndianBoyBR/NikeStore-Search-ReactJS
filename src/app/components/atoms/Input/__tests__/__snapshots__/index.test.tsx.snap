// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  position: relative;
}

.c1 {
  font-size: 1rem;
  padding: 0.9rem 2.5rem 0.9rem 0.5rem;
  outline: none;
  width: 100%;
  background-color: #FFFFFF;
  border: 1px solid #D6D6D6;
  color: #737373;
}

.c1::-webkit-search-decoration,
.c1::-webkit-search-cancel-button,
.c1::-webkit-search-results-button,
.c1::-webkit-search-results-decoration {
  display: none;
}

.c1::-webkit-input-placeholder {
  color: #737373;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #737373;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #737373;
  opacity: 1;
}

.c1::placeholder {
  color: #737373;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #737373;
}

<label
  class="c0"
>
  <input
    class="c1"
    data-testid="input"
    placeholder="placeholder"
    type="search"
  />
</label>
`;
